topi/python/topi/argwhere.py

```
@@ -0,0 +1,166 @@
+                        a[valid_index, 3] = i4
+                        valid_index += 1
+    return a
+
+@hybrid.script
+def hybrid_argwhere_5d(output_shape, condition):
+    """Find the indices of elements of a 5-D tensor that are non-zero.
+
+    Parameters
+    ----------
+    condition : tvm.Tensor
+        5-D tensor with boolean values.
+
+    Returns
+    -------
+    out : tvm.Tensor
+        Indices of non-zero elements.
+    """
+    a = output_tensor(output_shape, "int32")
+    a1 = condition.shape[0]
+    a2 = condition.shape[1]
+    a3 = condition.shape[2]
+    a4 = condition.shape[3]
+    a5 = condition.shape[4]
+    valid_index = 0
+    for i1 in range(a1):
+        for i2 in range(a2):
+            for i3 in range(a3):
+                for i4 in range(a4):
+                    for i5 in range(a5):
+                        if condition[i1, i2, i3, i4, i5]:
+                            a[valid_index, 0] = i1
+                            a[valid_index, 1] = i2
+                            a[valid_index, 2] = i3
+                            a[valid_index, 3] = i4
+                            a[valid_index, 4] = i5
+                            valid_index += 1
+    return a
+
+@tvm.target.generic_func
+def argwhere(output_shape, condition):
+    """Find the indices of elements of a tensor that are non-zero.
+
+    Parameters
+    ----------
+    condition : tvm.Tensor
+        Tensor with boolean values.
+
+    Returns
+    -------
+    out : tvm.Tensor
+        Indices of non-zero elements.
+    """
+    if len(condition.shape) == 2:
+        return hybrid_argwhere_2d(output_shape.shape, condition)
+    if len(condition.shape) == 3:
+        return hybrid_argwhere_3d(output_shape.shape, condition)
+    if len(condition.shape) == 4:
+        return hybrid_argwhere_4d(output_shape.shape, condition)
+    if len(condition.shape) == 5:
+        return hybrid_argwhere_5d(output_shape.shape, condition)
+    return []
```

We should assert failure here instead of returning an empty list.